ddtrace/internal/utils/http.py

```diff
@@ -38,6 +39,15 @@ def strip_query_string(url):
     return h + fs + f
 
 
+def redact_url(url, query_string_obfuscation_pattern):
+    # type: (str, re.Pattern) -> str
+    hqs, fs, f = url.partition("#")
+    h, qss, qs = hqs.partition("?")
+
+    redacted_query = query_string_obfuscation_pattern.sub("<redacted>", compat.to_unicode(qs))
```

Comment:
should we only call the regular expression if we have a query string present?

Reply:
done!